### PR TITLE
Register XLA lazy backend

### PR DESCRIPTION
**Pull request by JackCaoG**
In this pr we register the `XLABackend` to LTC and also register the `XLANativeFunctions` and `AutogradXLANativeFunctions` generated by LTC codegen. This helped us to get rid of the redundant codegen we have.

### CODE REVIEW

**Comment by JackCaoG** [comment]
@Krovatkin This pr also fixed the CI issue you had in your `ExecuteComputation` pr.

**Comment by JackCaoG** [comment]
Some CPP test seems to fail, will take a look tmr.

File: .gitignore [hunk 1]
```diff
@@ -10,11 +10,6 @@ torch_xla/csrc/version.cpp
 *.pyc
 *.so
 
-# Files autogenerated by scripts/generate_code.sh
```
Comment (wonjoolee95) [inline .gitignore:13]: Just curious -- why are we able to remove this? IIRC, we generate our xla ops each build. Do our existing ops still get generated if we remove this?

File: torch_xla/csrc/xla_backend_impl.cpp [hunk 2]
```diff
@@ -151,6 +158,10 @@ class XlaBackendImpl : public torch::lazy::BackendImplInterface {
     return at::DeviceType::CPU;
   }
 
+  void SetDefaultDeviceType(std::string type) override {
+    default_device_type_ = XlaDeviceType(c10::Device(type).type());
```
Comment (wonjoolee95) [inline torch_xla/csrc/xla_backend_impl.cpp:162]: nit: is the `c10::Device(type).type()` always same as our own definition of `XlaDeviceType` as defined in https://github.com/pytorch/xla/blob/master/torch_xla/csrc/device.h#L13?

**Review by wonjoolee95 — APPROVED** [review]
Thanks! Had some nit questions, but LGTM.

File: torch_xla/csrc/xla_backend_impl.cpp (hunk 2 above)
Comment (JackCaoG) [reply torch_xla/csrc/xla_backend_impl.cpp:162]: Ah OK, I guess the right thing to do here is to get a defualt device and set the default type correctly. In the TPU setup we actually want it to be TPU. I will update this.

Comment (JackCaoG) [reply torch_xla/csrc/xla_backend_impl.cpp:162]: hmm, wait. This function already takes string as type. What's the right thing to do here? How do I convert `string` to `XlaDeviceType`?

Maybe we need to convert  https://github.com/pytorch/xla/blob/dbc67604cfff305a37544513c20e377b254054ef/torch_xla/csrc/device.cpp#L53-L64

to a helper function? @wonjoolee95

Comment (wonjoolee95) [reply torch_xla/csrc/xla_backend_impl.cpp:162]: What does the `c10::Device(type).type()` return? Is it just the type of the device in string format like `TPU` or `CPU`? If so, I think you're right we can move this to a helper function.

**Pull request merged**